app/assets/javascripts/polls.js.coffee

```diff
@@ -0,0 +1,27 @@
+App.Polls =
+  generateToken: ->
+    token = ''
+    for n in [0..5]
+      rand = Math.random().toString(36).substr(2) # remove `0.`
```

Comment:
Could `rand` definition be outside the loop?

Reply:
nah its cool its not a function but a string my bad 😅

Reply:
Calling Math.random repeatedly is not a good way to collect entropy. Instead you may want to use crypto.getRandomValues. Here's an example  

```
// provides 128 bits of entropy
window.crypto.getRandomValues(new Uint32Array(4)).reduce(function(a, v) { return a.toString(36) + v.toString(36); })
```

You can calculate collision probabilities using the birthday attack model.